src/value.rs

```diff
@@ -42,6 +42,40 @@ impl Value {
             t: Scalar::zero(),
         }
     }
+
+    /// Creates variables for the fields in `Value`, and packages them in an `AllocatedValue`.
+    pub fn allocate<CS: ConstraintSystem>(&self, cs: &mut CS) -> Result<AllocatedValue, R1CSError> {
+        let (q_var, _, _) = cs.allocate(|| Ok((self.q.into(), Scalar::zero(), Scalar::zero())))?;
+        let (a_var, t_var, _) = cs.allocate(|| Ok((self.a, self.t, self.a * self.t)))?;
+
+        Ok(AllocatedValue {
+            q: q_var,
+            a: a_var,
+            t: t_var,
+            assignment: Some(*self),
+        })
+    }
+
+    pub fn allocate_empty<CS: ConstraintSystem>(cs: &mut CS) -> Result<AllocatedValue, R1CSError> {
```

Comment:
`allocate_unassigned` maybe?

Reply:
We already have "assignment" as a term, and it required me to look inside the code to understand what "empty" means.